### PR TITLE
Target Java 17 and Use GitHub Actions

### PR DESCRIPTION
**Pull request by aws-sde**
*Issue #, if available:*
N/A

*Description of changes:*
Dependencies already support JDK 17. This change actually targets 17 instead of 11.

It also replaces CodeBuild with GitHub Actions for CI.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-sde** [comment]
CI successfully ran in my fork but I can't get it to run in this PR. https://github.com/aws-sde/aws-cloudformation-resource-providers-transfer/actions/runs/8055721894/job/22003475949

Hoping we can override the CI and merge this PR once we have two approvals.

**Review by lydakis — APPROVED** [review]

**Review by vicwuqinyi — APPROVED** [review]

**Pull request merged**
